internal/controller/certificates/policies/checks.go

```diff
@@ -94,6 +94,33 @@ func SecretPrivateKeyMatchesSpec(input Input) (string, string, bool) {
 	return "", "", false
 }
 
+func SecretKeystoreFormatMatchesSpec(input Input) (string, string, bool) {
+	if input.Certificate.Spec.Keystores == nil {
+		if len(input.Secret.Data[cmapi.Pkcs12SecretKey]) == 0 ||
+			len(input.Secret.Data[cmapi.Pkcs12TruststoreKey]) == 0 ||
+			len(input.Secret.Data[cmapi.JksSecretKey]) == 0 ||
+			len(input.Secret.Data[cmapi.JksTruststoreKey]) == 0 {
+			return SecretMismatch, "Keystore keys does not contain data", true
+		}
+		return "", "", false
+	}
```

Comment:
Does cert-manager in fact delete these files from the target Secret when the keystore format changes?

It's unlikely, but worth thinking about what happens if the user has created the Secret themselves.
It the original Secret contained one of these file names and cert-manager *adopts* the Secret, 
is it correct for it to delete the files which it did not create?

Reply:
> Does cert-manager in fact delete these files from the target Secret when the keystore format changes?

Yes. If the keystore is switched from JKS to PKCS12, the jks trust store and key store are removed; pkcs12 trust store and keystore are added. Also, vice versa.

If we remove the keystore section from the certificate spec, all the keystore specific entries are removed from the secret.

> It's unlikely, but worth thinking about what happens if the user has created the Secret themselves.
It the original Secret contained one of these file names and cert-manager adopts the Secret,
is it correct for it to delete the files which it did not create?

Since these are cert-manager managed fields, it should be okay to delete them if keystore is not defined. Similar to other fields such as `ca.crt`, this is also managed by cert-manager.